code-dot-org/code-dot-org
# fix some "already initialized constant" warnings

sometimes, when you run unit tests or dashboard server, you get warnings like this:
```
Dave-MBP:~/src/cdo2/dashboard (staging)$ bundle exec spring testunit ./test/models/concerns/text_to_speech_test.rb
/Users/dsb/src/cdo2/dashboard/app/helpers/proxy_helper.rb:4: warning: already initialized constant SSL_HOSTNAME_MISMATCH_REGEX
/Users/dsb/src/cdo2/dashboard/app/helpers/proxy_helper.rb:4: warning: previous definition of SSL_HOSTNAME_MISMATCH_REGEX was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:8: warning: already initialized constant TTS_BUCKET
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:8: warning: previous definition of TTS_BUCKET was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:32: warning: already initialized constant TTSSafeRenderer
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:32: warning: previous definition of TTSSafeRenderer was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:34: warning: already initialized constant TTSSafeScrubber
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:34: warning: previous definition of TTSSafeScrubber was here
/Users/dsb/src/cdo2/dashboard/app/models/levels/dsl_defined.rb:247: warning: already initialized constant DslDefined
/Users/dsb/src/cdo2/dashboard/app/models/levels/dsl_defined.rb:247: warning: previous definition of DslDefined was here
/Users/dsb/src/cdo2/dashboard/app/models/unit.rb:36: warning: already initialized constant TEXT_RESPONSE_TYPES
/Users/dsb/src/cdo2/dashboard/app/models/unit.rb:36: warning: previous definition of TEXT_RESPONSE_TYPES was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:3: warning: already initialized constant IGNORED_SOLUTION_BLOCK_ATTRS
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:3: warning: previous definition of IGNORED_SOLUTION_BLOCK_ATTRS was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:10: warning: already initialized constant NEW_CATEGORY_XML
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:10: warning: previous definition of NEW_CATEGORY_XML was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:11: warning: already initialized constant STRIPPED_NODES_XPATH
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:11: warning: previous definition of STRIPPED_NODES_XPATH was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:12: warning: already initialized constant STRIPPED_ATTRS
/Users/dsb/src/cdo2/dashboard/app/models/concerns/solution_blocks.rb:12: warning: previous definition of STRIPPED_ATTRS was here
Running via Spring preloader in process 70558
```

This PR fixes some of these warnings. The remaining warnings are:
```
Dave-MBP:~/src/cdo2/dashboard (dave/fix-initialized-constant-warnings *)$ bundle exec spring testunit ./test/models/concerns/text_to_speech_test.rb
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:32: warning: already initialized constant TTSSafeRenderer
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:32: warning: previous definition of TTSSafeRenderer was here
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:34: warning: already initialized constant TTSSafeScrubber
/Users/dsb/src/cdo2/dashboard/app/models/concerns/text_to_speech.rb:34: warning: previous definition of TTSSafeScrubber was here
/Users/dsb/src/cdo2/dashboard/app/models/levels/dsl_defined.rb:247: warning: already initialized constant DslDefined
/Users/dsb/src/cdo2/dashboard/app/models/levels/dsl_defined.rb:247: warning: previous definition of DslDefined was here
Running via Spring preloader in process 69575
```

The remaining constants are being assigned more complex value types, making them riskier to change, thus warranting a separate PR.

## Testing story

* manually verified to produce above output as follows:
  * run a dashboard unit test using `spring testunit`
  * sync forward past a DB migration
  * run the migration on the test database
  * run the unit test again
* relying on drone to make sure I didn't break anything
